src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java

```diff
@@ -0,0 +1,255 @@
+        }
+        return invalidDaysOfWeek;
+    }
+
+    /**
+     * Checks whether the itinerary of a trip matches any of the OTP itineraries from the trip query params.
+     */
+    public void checkExistence() {
+        // TODO: Consider multi-threading?
+        // Check existence of itinerary in the response for each OTP request.
+        for (OtpRequest otpRequest : otpRequests) {
+            boolean hasMatchingItinerary = false;
+            DayOfWeek dayOfWeek = otpRequest.date.getDayOfWeek();
+            // Get existing result for day of week if a date for that day of week has already been processed, or create
+            // a new one.
+            ItineraryExistenceResult result = getResultForDayOfWeek(dayOfWeek);
+            if (result == null) {
+                result = new ItineraryExistenceResult();
+                setResultForDayOfWeek(result, dayOfWeek);
+            }
+            // Send off each plan query to OTP.
+            OtpDispatcherResponse response = OtpDispatcher.sendOtpPlanRequest(otpRequest);
+            TripPlan plan = response.getResponse().plan;
+            // Handle response if valid itineraries exist.
+            if (plan != null && plan.itineraries != null) {
+                for (Itinerary itineraryCandidate : plan.itineraries) {
+                    // If a matching itinerary is found, save the date with the matching itinerary.
+                    // The matching itinerary will replace the original trip.itinerary.
+                    // FIXME Replace 'equals' with matching itinerary
+                    if (itineraryCandidate.equals(referenceItinerary)) {
+                        result.handleValidDate(otpRequest.date, itineraryCandidate);
+                        hasMatchingItinerary = true;
+                    }
+                }
+            }
+            if (!hasMatchingItinerary) {
+                // If no match was found for the date, mark day of week as non-existent for the itinerary.
+                result.handleInvalidDate(otpRequest.date);
+            }
+        }
+        if (!this.allCheckedDaysAreValid()) {
+            this.message = String.format(
+                "The trip is not possible on the following days of the week you have selected: %s",
+                String.join(", ", getInvalidDaysOfWeek())
+            );
+            this.error = true;
+        }
+    }
+
+    /**
+     * Holds results for an itinerary existence check for a particular day of the week.
+     */
+    public static class ItineraryExistenceResult {
+        /**
+         * True if and only if an itinerary is available for all dates tested for existence.
+         */
+        @JsonProperty
+        public boolean isValid() {
+            return invalidDates.size() == 0;
+        }
+
+        /**
+         * Method and argument solely needed for accepting request contents that include a `valid` field
+         * (e.g. for checking a monitored trip with previous existence check record).
```

Comment:
This doesn't make a ton of sense to me.

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/77/commits/d26baf619770dce9e11667961912b80261de090c.